syntax = "proto3";

import "farm_ng_proto/tractor/v1/resource.proto";
import "google/protobuf/timestamp.proto";
package farm_ng_proto.tractor.v1;
option go_package = "github.com/farm_ng/genproto";

message CaptureCalibrationDatasetConfiguration {
  // Number of frames to capture
  int32 num_frames = 1;
  string name = 2;
}

message CaptureCalibrationDatasetStatus {
  // Number of frames captured
  int32 num_frames = 1;

  // The tag ids seen in the course of the capture
  repeated int32 tag_ids = 2;

  oneof input_required {
    CaptureCalibrationDatasetConfiguration input_required_configuration = 3;
  }

  CaptureCalibrationDatasetResult result = 4;
}

message CaptureCalibrationDatasetResult {
  // Number of frames captured
  int32 num_frames = 1;

  // The tag ids seen in the course of the capture
  repeated int32 tag_ids = 2;

  // The time the capture completed
  google.protobuf.Timestamp stamp_end = 3;

  // The calibration data, an IPC log, saved to disk in protobuf binary format.
  Resource dataset = 4;

  // This result, saved to disk in protobuf json format.
  Resource resource = 5;
}
